homeassistant/components/camera/webrtc.py

```diff
@@ -129,6 +210,85 @@ async def _async_refresh_providers(hass: HomeAssistant) -> None:
     )
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "camera/webrtc/offer",
+        vol.Required("entity_id"): cv.entity_id,
+        vol.Required("offer"): str,
+    }
+)
+@websocket_api.async_response
+async def ws_webrtc_offer(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle the signal path for a WebRTC stream.
+
+    This signal path is used to route the offer created by the client to the
+    camera device through the integration for negotiation on initial setup.
+    The ws endpoint returns an subscription id, where ice candidates and the
+    final answer will be returned.
+    The actual streaming is handled entirely between the client and camera device.
+
+    Async friendly.
+    """
+    entity_id = msg["entity_id"]
+    offer = msg["offer"]
+    camera = get_camera_from_entity_id(hass, entity_id)
+    if camera.frontend_stream_type != StreamType.WEB_RTC:
+        connection.send_error(
+            msg["id"],
+            "web_rtc_offer_failed",
+            (
+                "Camera does not support WebRTC,"
+                f" frontend_stream_type={camera.frontend_stream_type}"
+            ),
+        )
+        return
+
+    session_id = uuid.uuid4().hex
+    connection.subscriptions[msg["id"]] = partial(
+        camera.close_webrtc_session, session_id
+    )
+
+    # Send subscription id to client
+    connection.send_result(msg["id"], {"session_id": session_id})
```

Comment:
Subscriptions are not able to return data in the response to the subscription being established. The return value of a subscription message is an unsubscribe function: https://github.com/home-assistant/home-assistant-js-websocket/blob/master/lib/connection.ts#L348

Reply:
If you want the subscriber to know, send it as an event message.